src/Analyser/TypeSpecifier.php

```diff
@@ -818,6 +818,22 @@ public function specifyTypesInCondition(
 								$rootExpr,
 							),
 						);
+					} elseif ($var->dim instanceof Expr\Variable) {
+						$varType = $scope->getType($var->var);
+						$varIterableKeyType = $varType->getIterableKeyType();
+
+						if ($varIterableKeyType->isString()->yes() && $varIterableKeyType->isNumericString()->no()) {
+							$types = $types->unionWith(
+								$this->create(
+									$var->dim,
+									$varIterableKeyType,
+									$context,
+									false,
+									$scope,
+									$rootExpr,
+								),
+							);
+						}
```

Comment:
the more iterations I do on this things the more I feel I need a new `Type` method, e.g. `Type->toIssetOffset()`

Reply:
memo to me: maybe take inspiration from `array_key_exists`:

https://github.com/phpstan/phpstan-src/blob/e8027e7a59ba4e70925c89df51081902aa577e03/src/Type/Php/ArrayKeyExistsFunctionTypeSpecifyingExtension.php#L61-L93

Reply:
`array_key_exists` also needs improvements https://github.com/phpstan/phpstan/issues/11724 :)